### PR TITLE
added sort

### PR DESCRIPTION
**Pull request by NadezhdaGlin**

### CODE REVIEW

File: app/controllers/cargos_controller.rb [hunk 1]
```diff
@@ -1,8 +1,18 @@
...
     @cargos = Cargo.all
+=======
+  before_action :authenticate_user!
+  def index 
+    # sort = params.permit(:sort)
+    # sort_direction = params.permit(:sort_direction)
+    # binding.pry
+    # @cargos = current_user.cargos
+    @cargos = current_user.cargos.order("#{params[:sort]} #{params[:sort_direction]}") if params[:sort].present?
+>>>>>>> Stashed changes
```
Comment (NikolayGGorbunov) [inline app/controllers/cargos_controller.rb]: Clean up trash from stash

File: app/models/cargo.rb [hunk 2]
```diff
@@ -13,5 +13,11 @@ class Cargo < ApplicationRecord
             :origins,
             :destinations,
             presence: true
+<<<<<<< Updated upstream
   validates :phone, format: {with: /(^[\+]?[(]?[0-9]{3}[)]?[-\s\.]?[0-9]{3}[-\s\.]?[0-9]{4,6}$)/}
+=======
+  validates :phone, format: { with: /(^\+?[(]?[0-9]{3}[)]?[-\s.]?[0-9]{3}[-\s.]?[0-9]{4,6}$)/ }
+  belongs_to :user
+
+>>>>>>> Stashed changes
```
Comment (NikolayGGorbunov) [inline app/models/cargo.rb]: Same as above

File: app/views/cargos/index.html.erb [hunk 3]
```diff
@@ -1,4 +1,10 @@
 <h1>ORDERS</h1>
+<<<<<<< Updated upstream
+=======
+
+<%= button_to "Sorted by price", {controller: :cargos, action: :index}, method: :get, params: {sort: 'price', sort_direction: 'desc'} %>
+
+>>>>>>> Stashed changes
```
Comment (NikolayGGorbunov) [inline app/views/cargos/index.html.erb]: Same as above

File: spec/controllers/cargos_controller_spec.rb [hunk 4]
```diff
@@ -31,16 +32,34 @@
...
+      expect(index).to have_http_status(200)
+    end
+  end
+
+  context '#show' do
+    subject(:show) { get :show, params: { id: Cargo.last.id } }
+    it 'returns cargo' do
+      show
+      expect(assigns(:cargo)).to eq(Cargo.last)
+>>>>>>> Stashed changes
```
Comment (NikolayGGorbunov) [inline spec/controllers/cargos_controller_spec.rb]: Same as above

File: spec/controllers/cargos_controller_spec.rb [hunk 5]
```diff
@@ -49,22 +68,24 @@
...
           expect{create_cargo}.to change(Cargo, :count).by(1)
         end
+=======
+  context '#create' do
+    let!(:cargo_values) { attributes_for :cargo }
+    subject(:create_cargo) { post :create, params: { cargo: cargo_values } }
+    it 'return created cargo' do
+      VCR.use_cassette('Cargo_controller') do
+        expect { create_cargo }.to change(Cargo, :count).by(1)
+>>>>>>> Stashed changes
```
Comment (NikolayGGorbunov) [inline spec/controllers/cargos_controller_spec.rb]: Same as above

File: spec/controllers/cargos_controller_spec.rb [hunk 6]
```diff
@@ -49,22 +68,24 @@
...
+        expect(create_cargo).to redirect_to(cargo_path(Cargo.last))
       end
     end
-
-    context 'Status 302' do
-      it 'return status' do
-        VCR.use_cassette('Cargo_controller') do
-          expect(create_cargo).to have_http_status(302)
-        end
+    it 'return status' do
```
Comment (NikolayGGorbunov) [inline spec/controllers/cargos_controller_spec.rb]: What status it should return?

File: spec/controllers/cargos_controller_spec.rb [hunk 7]
```diff
@@ -28,64 +27,115 @@
         expect(assigns(:cargos)).to eq(Cargo.all.order(price: :desc))
```
Comment (NikolayGGorbunov) [inline spec/controllers/cargos_controller_spec.rb]: In Cargo.all may presence other user's cargos.

File: spec/controllers/cargos_controller_spec.rb [hunk 8]
```diff
@@ -11,12 +11,11 @@
   let!(:cargo2) { create :cargo, user_id: user.id, price: 2643 }
 
   context '#index' do
```
Comment (NikolayGGorbunov) [inline spec/controllers/cargos_controller_spec.rb:13]: Maybe we should add tests for multiply users

File: spec/controllers/cargos_controller_spec.rb [hunk 9]
```diff
@@ -28,64 +27,115 @@
...
+      it 'return status 302' do
+        expect(index).to have_http_status(302)
+      end
     end
   end
 
   context '#show' do
+    subject(:show) { get :show, params: { id: Cargo.last.id } }
 
     context 'when user authorized' do
```
Comment (NikolayGGorbunov) [inline spec/controllers/cargos_controller_spec.rb:65]: What happen if user try to see other user's package?

File: spec/controllers/cargos_controller_spec.rb (hunk 6 above)
Comment (NadezhdaGlin) [reply spec/controllers/cargos_controller_spec.rb]: fixed

File: spec/controllers/cargos_controller_spec.rb (hunk 5 above)
Comment (NadezhdaGlin) [reply spec/controllers/cargos_controller_spec.rb]: fixed

File: spec/controllers/cargos_controller_spec.rb (hunk 4 above)
Comment (NadezhdaGlin) [reply spec/controllers/cargos_controller_spec.rb]: fixed

File: spec/controllers/cargos_controller_spec.rb [hunk 10]
```diff
@@ -28,64 +27,115 @@
...
       before { sign_in user }
-      let!(:cargo_values) { attributes_for :cargo }
-      subject(:create_cargo) { post :create, params: { cargo: cargo_values } }
 
       it 'return created cargo' do
         VCR.use_cassette('Cargo_controller') do
           expect { create_cargo }.to change(Cargo, :count).by(1)
         end
       end
```
Comment (NikolayGGorbunov) [inline spec/controllers/cargos_controller_spec.rb]: This 'it' doesn't return cargo

File: app/views/cargos/index.html.erb (hunk 3 above)
Comment (NadezhdaGlin) [reply app/views/cargos/index.html.erb]: fixed

File: app/models/cargo.rb (hunk 2 above)
Comment (NadezhdaGlin) [reply app/models/cargo.rb]: fixed

File: app/controllers/cargos_controller.rb (hunk 1 above)
Comment (NadezhdaGlin) [reply app/controllers/cargos_controller.rb]: fixed

File: spec/controllers/cargos_controller_spec.rb (hunk 10 above)
Comment (NadezhdaGlin) [reply spec/controllers/cargos_controller_spec.rb]: you mean, 'it' should it be written differently?

File: spec/controllers/cargos_controller_spec.rb (hunk 9 above)
Comment (NadezhdaGlin) [reply spec/controllers/cargos_controller_spec.rb:65]: but user does not see other people's cargos

File: spec/controllers/cargos_controller_spec.rb (hunk 8 above)
Comment (NadezhdaGlin) [reply spec/controllers/cargos_controller_spec.rb:13]: multiple users?

Comment (NikolayGGorbunov) [reply spec/controllers/cargos_controller_spec.rb:13]: Yes. Does #index assign packages from other users?

File: spec/controllers/cargos_controller_spec.rb (hunk 9 above)
Comment (NikolayGGorbunov) [reply spec/controllers/cargos_controller_spec.rb:65]: Yes, but we can send request manually into a URL

File: spec/controllers/cargos_controller_spec.rb (hunk 10 above)
Comment (NikolayGGorbunov) [reply spec/controllers/cargos_controller_spec.rb]: Yes, 'increase cargos count' or 'creates cargo' for example

**Review by NikolayGGorbunov — APPROVED** [review]

**Pull request merged**